controllers/component_image_controller_test.go

```diff
@@ -435,6 +435,98 @@ var _ = Describe("Component image controller", func() {
 			Expect(repoImageInfo.Secret).To(Equal(resourceKey.Name))
 		})
 
+		It("should stop and report error if image repository creation fails", func() {
+			isCreateRepositoryInvoked := false
+			CreateRepositoryFunc = func(repository quay.RepositoryRequest) (*quay.Repository, error) {
+				isCreateRepositoryInvoked = true
+				return nil, fmt.Errorf("fail to marshal data")
+			}
+
+			repositoryInfoJsonBytes, _ := json.Marshal(ImageRepositoryStatus{})
+			setComponentAnnotationValue(resourceKey, ImageAnnotationName, string(repositoryInfoJsonBytes))
+			setComponentAnnotationValue(resourceKey, GenerateImageAnnotationName, `{"visibility": "public"}`)
+
+			Eventually(func() bool { return isCreateRepositoryInvoked }, timeout, interval).Should(BeTrue())
+
+			expectedValue, _ := json.Marshal(&ImageRepositoryStatus{Message: "failed to generate image repository"})
+			waitComponentAnnotationWithValue(resourceKey, ImageAnnotationName, string(expectedValue))
+		})
+
+		It("should do nothing and set error for changing visibility if image is invalid in image annotation", func() {
+			CreateRepositoryFunc = func(repository quay.RepositoryRequest) (*quay.Repository, error) {
+				defer GinkgoRecover()
+				Fail("Image repository creation should not be invoked")
+				return nil, nil
+			}
+
+			// An invalid image is set, which does not include registry.
+			setComponentAnnotationValue(resourceKey, ImageAnnotationName, `{"image": "ns/img:tag", "secret": "1234"}`)
+			setComponentAnnotationValue(resourceKey, GenerateImageAnnotationName, `{"visibility": "private"}`)
+
+			waitComponentAnnotationGone(resourceKey, GenerateImageAnnotationName)
+			waitComponentAnnotation(resourceKey, ImageAnnotationName)
+
+			repoImageInfo := &ImageRepositoryStatus{}
+			component := getComponent(resourceKey)
+			Expect(json.Unmarshal([]byte(component.Annotations[ImageAnnotationName]), repoImageInfo)).To(Succeed())
+			Expect(repoImageInfo.Message).To(Equal("Invalid image url"))
+		})
+
+		It("should stop if fail to change repository visibility", func() {
```

Comment:
Looks like a duplicate of `should add message and stop if it's not possible to switch image repository visibility`

Reply:
It covers this `else` branch https://app.codecov.io/gh/redhat-appstudio/image-controller/pull/55/blob/controllers/component_image_controller.go#L223

Reply:
Then the test name is incorrect, since it doesn't stop, but retries to change visibility.